heim-runtime/src/linux.rs

```diff
@@ -0,0 +1,27 @@
+use once_cell::sync::OnceCell;
+use std::path::PathBuf;
+
+type RootLock = OnceCell<PathBuf>;
+
+static PROCFS_ROOT: RootLock = OnceCell::new();
+static SYSFS_ROOT: RootLock = OnceCell::new();
+
+pub fn set_procfs_root(root: &str) {
+    let _ = PROCFS_ROOT.get_or_init(|| PathBuf::from(root));
+}
+
+pub fn procfs_root() -> PathBuf {
+    PROCFS_ROOT
+        .get_or_init(|| PathBuf::from("/proc"))
+        .to_path_buf()
+}
+
+pub fn set_sysfs_root(root: &str) {
+    let _ = SYSFS_ROOT.get_or_init(|| PathBuf::from(root));
+}
+
+pub fn sysfs_root() -> PathBuf {
```

Comment:
We need to change these function signatures according to this snippet: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=db5699463386bfdc45566dddf581edba
Originally posted by @MOZGIII in this comment: https://github.com/heim-rs/heim/pull/289#issuecomment-740219494